src/Config/ConfigLocator.php

```diff
@@ -256,29 +258,116 @@ public function addDrushConfig($drushProjectDir)
      * Add any configuration files found around the Drupal root of the
      * selected site.
      *
-     * @param Path to the selected Drupal site
+     * @param $drupalRoot
+     *   Path to the selected Drupal site.
      * @return $this
      */
-    public function addSitewideConfig($siteRoot)
+    public function addDrupalConfig($drupalRoot)
     {
         // There might not be a site.
-        if (!is_dir($siteRoot)) {
+        if (!is_dir($drupalRoot)) {
             return;
         }
 
         // We might have already processed this root.
-        $siteRoot = realpath($siteRoot);
-        if (in_array($siteRoot, $this->siteRoots)) {
+        $drupalRoot = realpath($drupalRoot);
+        if (in_array($drupalRoot, $this->drupalRoots)) {
             return;
         }
 
         // Remember that we've seen this location.
-        $this->siteRoots[] = $siteRoot;
+        $this->drupalRoots[] = $drupalRoot;
 
-        $this->addConfigPaths(self::DRUPAL_CONTEXT, [ dirname($siteRoot) . '/drush', "$siteRoot/drush", "$siteRoot/sites/all/drush" ]);
+        $this->addConfigPaths(self::DRUPAL_CONTEXT, [ dirname($drupalRoot) . '/drush', "$drupalRoot/drush", "$drupalRoot/sites/all/drush" ]);
         return $this;
     }
 
+    /**
+     * Add any configuration files found around the multisite directory.
+     *
+     * @param string $drupalRoot
+     *   Path to the selected Drupal site.
+     *
+     * @param \Drush\SiteAlias\AliasRecord $alias
+     *   Site URI of the multisite.
+     *
+     * @return $this
+     */
+    public function addSiteConfig($drupalRoot, $alias)
+    {
+        $uri = $alias->uri() ?: 'default';
+
+        // Convert a fqdn to a hostname and look for matching entry in
+        // sites/sites.php.
+        if (filter_var($uri, FILTER_VALIDATE_URL)) {
+            $hostname = $this->convertUriToHostname($uri);
+
+            // If $hostname matches a sites.php mappings, use dir from mapping.
+            if ($dir_name = $this->lookupSiteDirFromHostname($hostname, $drupalRoot)) {
+                $uri = $dir_name;
+            }
+        }
+
+        // There might not be a site directory.
+        $site_dir = "$drupalRoot/sites/$uri";
```

Comment:
Maybe site dir lookup should be a method of the alias record class (bring along convertUriToHostname, lookupSiteDirFromHostname).

Reply:
n.b. I'm not sure about this - considering it. Maybe factoring out into a utility class would be better. I don't think that I want the alias record code to know about Drupal site directory conventions. Ideally the site alias classes will be factored out into a separate consolidation project.